common/ovf_deploy.yml

```diff
@@ -21,3 +21,34 @@
 - name: Display the result of ovf template deploy
   ansible.builtin.debug: var=ovf_deploy
   when: enable_debug is defined and enable_debug
+
+- name: "Update VM memory"
+  when: memory_mb is defined and memory_mb
+  block:
+    - name: "Set the facts of vm memory and initial configuration"
+      ansible.builtin.set_fact:
+        vm_mem_mb: "{{ ovf_deploy.instance.hw_memtotal_mb | int }}"
+        initial_mem_mb: "{{ memory_mb | int }}"
+    - name: "Set initial VM memory"
+      include_tasks: vm_set_memory_size.yml
+      vars:
+        memory_mb: "{{ initial_mem_mb }}"
+      when: vm_mem_mb | int != initial_mem_mb | int
```

Comment:
I think above set_fact task is not necessary. Here can directly check `ovf_deploy.instance.hw_memtotal_mb | int != memory_mb | int`. And the `block` and `vars` can be removed.